src/react-hooks/buildHooks.ts

```diff
@@ -262,6 +322,25 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
     return {
       useQueryState,
       useQuerySubscription,
+      useLazyQuerySubscription,
+      useLazyQuery(options) {
+        const [providedArgs, setProvidedArgs] = useState(null);
+        const [trigger, lastPromise] = useLazyQuerySubscription(options);
+        const queryStateResults = useQueryState(providedArgs, {
+          ...options,
+          skip: providedArgs === null,
+        });
+        const triggerQuery = useCallback(
+          function (args: any) {
+            setProvidedArgs(args);
+            trigger(args);
+          },
+          [trigger]
+        );
+
+        const info = useMemo(() => ({ lastArgs: lastPromise?.arg }), [lastPromise]);
+        return useMemo(() => [triggerQuery, queryStateResults, info], [triggerQuery, queryStateResults, info]);
+      },
```

Comment:
I still have a concern here. It doesn't make sense for someone to call a trigger like: `getPosts(null)`, but if they did, this would be broken :thinking: . Maybe we should use some type of key like `__rtkq/uninitialized` instead `null`? Not too sure. @phryneas @markerikson any thoughts here?